compiler/sempass2.nim

```diff
@@ -259,6 +269,33 @@ proc listGcUnsafety(s: PSym; onlyWarning: bool; conf: ConfigRef) =
   var cycleCheck = initIntSet()
   listGcUnsafety(s, onlyWarning, cycleCheck, conf)
 
+proc listSideEffects(s: PSym; onlyWarning: bool; cycleCheck: var IntSet; conf: ConfigRef) =
+  for u in s.sideEffectReasons:
```

Comment:
seems expensive; is that called in a compiles() context? if so, it needs to be disabled in the same way as extended diagnostics are skipped for sigmatch